### PR TITLE
Socket optimization

### PR DESCRIPTION
**Pull request by ZynoXelek**
Client now uses a single socket (persistent) to communicate with the server.
The server handles multiple sockets and save them in order to communicate with each client.

Still need to make the server use the sockets to identify players instead of saving the player ip in the Player class.

### CODE REVIEW

**Review by Shisohka — APPROVED** [review]

**Review by BlueNZ — COMMENTED** [review]
Should wait for wall display pull request to be accepted

File: server.py [hunk 1]
```diff
@@ -209,44 +230,161 @@ def colorNewPlayer():
...
+                
+                print("Socket server closed !")
+                
+                for (username,(sock,addr)) in dicoSocket:
+                    sock.shutdown(SHUT_RDWR)
+                    sock.close()
+                
+                print("Client sockets closed !")
+            case "deaf":
+                LISTENING = False
```
Comment (Laudut) [inline server.py:258]: we need to add " .shutdown(...)"

**Review by Laudut — APPROVED** [review]

File: server.py (hunk 1 above)
Comment (Laudut) [reply server.py:258]: but maybe we can do that later

Comment (ZynoXelek) [reply server.py:258]: Yes we will have to find how to do it !
I quickly tried to add a shutdown call here but it did not go very well during my tests...
There must be another way to do so but I don't know yet how to do it.

Though we will have to find a solution as we must be able to have a maximum number of players and that the easiest way to do so would be to stop listening from new connections I think.

**Comment by ZynoXelek** [comment]
Now that the wall fix has been pushed on main, conflicts from the update function in the player class were resolved.

**Comment by ZynoXelek** [comment]
Missclicked...

**Review by ZynoXelek — COMMENTED** [review]
Now that the wall fix has been pushed on main, conflicts from the update function in the player class were resolved.

File: server.py (hunk 1 above)
Comment (ZynoXelek) [reply server.py:258]: We could eventually use a
```
if LISTENING:
    out = process_request(...)
```
after the `MAINSOCKET.accept()`

Comment (ZynoXelek) [reply server.py:258]: We should definitely look closer at this : https://stackoverflow.com/questions/10090236/how-to-stop-a-python-socket-accept-call
I'll do it another day.

**Review by BlueNZ — APPROVED** [review]
If it works it works

**Review by Laudut — APPROVED** [review]

**Pull request merged**